### PR TITLE
File receive endpoint

### PR DESCRIPTION
**Pull request by steplica**
This will let the AlignmentController receive files as a part of the JSON payload. It will also remove some unnecessary services, converting them to util files instead.

### CODE REVIEW

File: src/main/java/org/moltimate/moltimatebackend/request/ActiveSiteAlignmentRequest.java [hunk 1]
```diff
@@ -13,8 +21,22 @@
...
 @AllArgsConstructor
-public class ActiveSiteAlignmentRequest extends AlignmentRequest {
+public class ActiveSiteAlignmentRequest {
 
-    private List<String> pdbIds;
-    private String ecNumber;
+    private List<String> pdbIds = new ArrayList<>();
+    private List<String> options = new ArrayList<>();
+    private List<String> filters = new ArrayList<>();
+    private List<MultipartFile> files = new ArrayList<>(); // TODO: Rename to customMotifs
```
Comment (jmiller656) [inline src/main/java/org/moltimate/moltimatebackend/request/ActiveSiteAlignmentRequest.java]: Are these TODOs for later?

File: src/main/java/org/moltimate/moltimatebackend/service/ActiveSiteService.java [hunk 2]
```diff
@@ -22,7 +21,6 @@
  * active site table.
  */
 @Service
-@Validated
```
Comment (jmiller656) [inline src/main/java/org/moltimate/moltimatebackend/service/ActiveSiteService.java]: What does this do?

File: src/main/java/org/moltimate/moltimatebackend/service/AlignmentService.java [hunk 3]
```diff
@@ -30,39 +30,36 @@
...
      */
-    public AlignmentResponse alignActiveSites(ActiveSiteAlignmentRequest alignmentRequest) {
-        return alignActiveSites(
-                proteinService.queryPdb(alignmentRequest.getPdbIds()),
-                alignmentRequest.getEcNumber()
-        );
-    }
+    public ActiveSiteAlignmentResponse alignActiveSites(ActiveSiteAlignmentRequest alignmentRequest) {
+        List<Structure> sourceStructures = alignmentRequest.getPdbIdsAsStructures();
+        List<Motif> customMotifs = alignmentRequest.getCustomMotifs();
```
Comment (jmiller656) [inline src/main/java/org/moltimate/moltimatebackend/service/AlignmentService.java]: Should we be paging these too?

File: src/main/java/org/moltimate/moltimatebackend/util/FileUtils.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    private static final PDBFileReader PDB_FILE_READER = new PDBFileReader();
+    private static final MMCIFFileReader MMCIF_FILE_READER = new MMCIFFileReader();
+
+    public static Motif getMotifFromFile(MultipartFile file) {
+        return null;
+
+        // TODO: Read motif metadata from file and process it
+
+//        InputStream fileInputStream;
+//        try {
```
Comment (jmiller656) [inline src/main/java/org/moltimate/moltimatebackend/util/FileUtils.java]: Changing this later?

File: src/main/java/org/moltimate/moltimatebackend/service/ActiveSiteService.java (hunk 2 above)
Comment (steplica) [reply src/main/java/org/moltimate/moltimatebackend/service/ActiveSiteService.java]: It's an annotation that lets entities/embeddables have certain properties be validated based on constraints. I don't know how it made its way into a service, it shouldn't have been there.

File: src/main/java/org/moltimate/moltimatebackend/service/AlignmentService.java (hunk 3 above)
Comment (steplica) [reply src/main/java/org/moltimate/moltimatebackend/service/AlignmentService.java]: We shouldn't paginate custom motifs, but I can see an argument for paginating source structures. It's out of scope for this project iteration and would only be an optimization on enormous queries (thousands of PDB Ids)

File: src/main/java/org/moltimate/moltimatebackend/request/ActiveSiteAlignmentRequest.java (hunk 1 above)
Comment (steplica) [reply src/main/java/org/moltimate/moltimatebackend/request/ActiveSiteAlignmentRequest.java]: Thanks! Made the change.
The ecNumber TODO is for later

File: src/main/java/org/moltimate/moltimatebackend/util/FileUtils.java (hunk 4 above)
Comment (steplica) [reply src/main/java/org/moltimate/moltimatebackend/util/FileUtils.java]: Thanks! I removed the commented code and left the TODO

File: src/main/java/org/moltimate/moltimatebackend/request/ActiveSiteAlignmentRequest.java (hunk 1 above)
Comment (jmiller656) [reply src/main/java/org/moltimate/moltimatebackend/request/ActiveSiteAlignmentRequest.java]: 👍

File: src/main/java/org/moltimate/moltimatebackend/service/ActiveSiteService.java (hunk 2 above)
Comment (jmiller656) [reply src/main/java/org/moltimate/moltimatebackend/service/ActiveSiteService.java]: 👍

File: src/main/java/org/moltimate/moltimatebackend/service/AlignmentService.java (hunk 3 above)
Comment (jmiller656) [reply src/main/java/org/moltimate/moltimatebackend/service/AlignmentService.java]: Sounds good!

File: src/main/java/org/moltimate/moltimatebackend/util/FileUtils.java (hunk 4 above)
Comment (jmiller656) [reply src/main/java/org/moltimate/moltimatebackend/util/FileUtils.java]: 👍

**Comment by jmiller656** [comment]
LGTM

**Pull request merged**